frontend/src/context/UserContext.tsx

```diff
@@ -0,0 +1,63 @@
+/* eslint-disable react-refresh/only-export-components */
+import {
+  createContext,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+import { InfoResponse, UserInfo } from "../types";
+import axios from "axios";
+
+interface UserContextType {
+  info: UserInfo | null;
+  loading: boolean;
+  logout: () => void;
+}
+
+const UserContext = createContext({} as UserContextType);
+
+export function UserProvider({ children }: { children: React.ReactNode }) {
+  const [info, setInfo] = useState<UserContextType["info"]>(null);
+  const [loading, setLoading] = useState(true);
+
+  const logout = async () => {
+    try {
+      await axios.get('/logout');
+    } catch (error) {
+      // TODO: Handle failure
+      console.error('Error during logout:', error);
+    }
+    setInfo(null);
+  };
+
+  const getInfo = async () => {
+    try {
+      const response = await axios.get('/api/info');
+      const data: InfoResponse = response.data;
+      setInfo(data);
+    } catch (error) {
+      console.error('Error getting user info:', error);
+    } finally {
+      setLoading(false);
+    }
+  }
+
+  // Invoked only on on mount.
+  useEffect(() => {
```

Comment:
If we used SWR we could simplify this a lot.
It would just be something like `const {data: info, loading} = useSWR<Info>("/api/info");`

Reply:
Leaving this for you @rolznz